extensions/reactive-mssql-client/deployment/src/test/java/io/quarkus/reactive/mssql/client/ChangingCredentialsTestResource.java

```diff
@@ -0,0 +1,41 @@
+package io.quarkus.reactive.mssql.client;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import jakarta.annotation.PostConstruct;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+
+import io.smallrye.mutiny.Uni;
+import io.vertx.mutiny.mssqlclient.MSSQLPool;
+
+@Path("/test")
+public class ChangingCredentialsTestResource {
+
+    @Inject
+    MSSQLPool client;
+
+    @Inject
+    ChangingCredentialsProvider credentialsProvider;
+
+    @PostConstruct
+    void addUser() {
+        client.query("CREATE LOGIN user2 WITH PASSWORD = 'user2_Has_A_Str0ng_Required_Password'").executeAndForget();
```

Comment:
There is a few issues here. `executeAndForget` won't wait for the execution, meaning the the other methods can be called while this one is still in progress. 
I think you need to make sure it's completed before. But now there is a problem as the method is likely going to be invoked on the event loop.

I would recommend switching to an `public void setup(@Observes StartupEvent)` which is called on a worker thread.

Reply:
In fact, it is guaranteed that this query will be executed before any other because the pool has a maximum size of 1 connection.

Nevertheless, you are right, using a setup method makes it more clear we want this to be executed before starting the test.